contrib/cirrus/cache_release.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+set -e
+
+source $(dirname $0)/lib.sh
+
+req_env_var GOSRC CIRRUS_BUILD_ID
+
+cd $GOSRC
+
+make release_info.txt
+FIELDS="$RELEASE_BASENAME $RELEASE_VERSION $RELEASE_DIST $RELEASE_DIST_VER $RELEASE_ARCH $RELEASE_FILE"
+cat release_info.txt | read "$FIELDS"
+req_env_var $FIELDS
+
+if [[ "$CI" == "true" ]]
+then
+    req_env_var CIRRUS_BRANCH CIRRUS_TASK_NAME CIRRUS_PR
```

Comment:
Will `CIRRUS_PR` be set if `CIRRUS_BRANCH` == master? (This may be a dumb question; Cirrus documentation can be daunting)

Reply:
My understanding is `CIRRUS_PR` is either NOT set or not reliable when `CIRRUS_BRANCH == master`.  For sure in PRs, `CIRRUS_BRANCH=pull/<number>`, and that seems mostly reliable.  

Yes the documentation can be a big confusing, none of the engineers are native-English speakers.  Why I want to get this to a semi-functional prototype stage first.

Reply:
That's why I asked: perhaps it might be better to `req_env_var CIRRUS_PR` lower down, when it's needed, instead of risking an unnecessary (and misleading) failure in this spot.

Reply:
I checked a few current and past runs.  CIRRUS_PR is always empty when running against a branch.

Reply:
I don't think I've expressed myself properly. If, as you say, CIRRUS_PR is empty in any real-world circumstance, then isn't it a very bad idea to `req_env_var` it? The code I (still) see does the req_env_var, then immediately checks it with `[[ -n ]]`.

Reply:
ohhh :relaxed: indeed a terrible idea, thx, I'll fix.

Reply:
okay fixed for real.